ignite/distributed/comp_models/native.py

```diff
@@ -0,0 +1,204 @@
+        all_env_vars_defined = [k in os.environ for k in necessary_env_vars]
+        if any(all_env_vars_defined) and not all(all_env_vars_defined):
+            raise RuntimeError(
+                "PyTorch distributed configuration should define env variables '{}'".format(necessary_env_vars)
+            )
+
+        os.environ["RANK"] = os.environ.get("RANK", "0")
+        os.environ["LOCAL_RANK"] = os.environ.get("LOCAL_RANK", "0")
+        os.environ["WORLD_SIZE"] = os.environ.get("WORLD_SIZE", "1")
+        os.environ["MASTER_ADDR"] = os.environ.get("MASTER_ADDR", "127.0.0.1")
+        os.environ["MASTER_PORT"] = os.environ.get("MASTER_PORT", "15000")
+
+    def _setup_env_in_slurm(self):
+        for k in ["SLURM_PROCID", "SLURM_LOCALID", "SLURM_NTASKS", "SLURM_JOB_NODELIST"]:
+            if k not in os.environ:
+                raise RuntimeError("SLURM distributed configuration is missing '{}' in env variables".format(k))
+
+        os.environ["RANK"] = os.environ["SLURM_PROCID"]
+        os.environ["LOCAL_RANK"] = os.environ["SLURM_LOCALID"]
+        os.environ["WORLD_SIZE"] = os.environ["SLURM_NTASKS"]
+        # port should be the same over all process
+        slurm_port = os.environ["SLURM_JOB_ID"]
+        slurm_port = slurm_port[-4:]
+        os.environ["MASTER_PORT"] = str(int(slurm_port) + 15000)
+        # master address is the first hostname of nodes list
+        hostnames = subprocess.check_output(["scontrol", "show", "hostnames", os.environ["SLURM_JOB_NODELIST"]])
+        os.environ["MASTER_ADDR"] = hostnames.split()[0].decode("utf-8")
+
+    def get_local_rank(self) -> int:
+        return self._local_rank
+
+    def get_rank(self) -> int:
+        return dist.get_rank()
+
+    def get_world_size(self) -> int:
+        return dist.get_world_size()
+
+    def get_ntasks_per_node(self) -> int:
+        return self._ntasks_per_node
+
+    def get_num_nodes(self) -> int:
+        return self._nnodes
+
+    def get_node_rank(self) -> int:
+        return self._node
+
+    def device(self) -> Union[torch.device, str]:
+        if self.backend() == "nccl":
+            return "cuda:{}".format(torch.cuda.current_device())
+        return "cpu"
+
+    def backend(self) -> Optional[str]:
+        return dist.get_backend()
+
+    def finalize(self):
+        dist.destroy_process_group()
+
+    @staticmethod
+    def _dist_worker_task_fn(
+        local_rank, backend, fn, world_size, num_procs_per_node, node_rank, master_addr, master_port, args
+    ):
+        from ignite.distributed.utils import _set_model
+
+        os.environ["LOCAL_RANK"] = str(local_rank)
+        os.environ["RANK"] = str(node_rank * num_procs_per_node + local_rank)
+        os.environ["WORLD_SIZE"] = str(world_size)
+        os.environ["MASTER_ADDR"] = str(master_addr)
+        os.environ["MASTER_PORT"] = str(master_port)
+
+        model = _NativeDistModel.create_from_backend(backend)
+        _set_model(model)
+        fn(local_rank, *args)
```

Comment:
`local_rank` should not be mandatory in user function. `idiot.get_local_rank()` should the preferred way to get the local rank.

Reply:
We pass local rank to be compatible with torch multiprocessing: https://pytorch.org/docs/stable/multiprocessing.html#torch.multiprocessing.spawn

Reply:
Ok I understand even if I would have preferred that, by default, there was no constraint on the signature of the user method.